nucypher/acumen/perception.py

```diff
@@ -62,29 +108,33 @@ def __setitem__(self, checksum_address, node_or_sprout):
             self.log.warn(msg)
 
     def __getitem__(self, checksum_address):
-        return self._nodes[checksum_address]
+        return self.__nodes[checksum_address]
 
     def __bool__(self):
-        return bool(self._nodes)
+        return bool(self.__nodes)
 
     def __contains__(self, item):
-        return item in self._nodes.keys() or item in self._nodes.values()
+        return item in self.__nodes.keys() or item in self.__nodes.values()
 
     def __iter__(self):
-        yield from self._nodes.values()
+        yield from self.__nodes.values()
```

Comment:
So, despite the bucketing, we still return all the nodes? Would it be better to exclude the bucketed ones here and not wait for pruning to get around to it?

Reply:
Tough to answer right now.  I think we can remove this method in favor of `FleetSensor.get_nodes`...  This largely depends on what public facing perspective on peers is.  Some buckets are positive health indicators, like the `VERIFIED` bucket - and perhaps some operations (like granting) can use a view of the network constrained on this basis.

Reply:
> I think we can remove this method in favor of `FleetSensor.get_nodes`...

I would actually be in favor of that. I think `FleetSensor` is too complicated of a class (especially with the addition of buckets) to be used as a simple collection.